include/zephyr/net/net_compat.h

```diff
@@ -298,10 +298,6 @@ extern "C" {
 #define IN6_IS_ADDR_MC_SITELOCAL(addr)  ZSOCK_IN6_IS_ADDR_MC_SITELOCAL(addr)
 #define IN6_IS_ADDR_MC_ORGLOCAL(addr)   ZSOCK_IN6_IS_ADDR_MC_ORGLOCAL(addr)
 
-#if !defined(ARPHRD_ETHER)
-#define ARPHRD_ETHER NET_ARPHRD_ETHER
-#endif
-
```

Comment:
I was getting an warning (promoted to error) about this being redefined elsewhere, and this seemed to fix the issue.